blurring_as_a_service/inference_pipeline/submit_inference_pipeline.py

```diff
@@ -13,61 +14,64 @@
 
 @pipeline()
 def inference_pipeline(workspace_name, subscription_id, resource_group):
-    # Iterate over all customers
-    for customer in inference_settings['customers']:
-        customer_name = customer['name']
-
-        # Format the root path of the Blob Storage Container in Azure using placeholders
-        blob_container_path = customer['container_root'].format(
-            subscription=subscription_id,
-            resourcegroup=resource_group,
-            workspace=workspace_name,
-            datastore_name=f"{customer_name}_input_structured"
-        )
-
-        input_root_folder = Input(
-            type=AssetTypes.URI_FOLDER,
-            path=blob_container_path,
-            description="Data to be blurred",
-        )
-
-        # Get the txt file that contains all paths of the files to run inference on
-        files_to_blur_path = customer['inputs']['files_to_blur'].format(
-            subscription=subscription_id,
-            resourcegroup=resource_group,
-            workspace=workspace_name,
-            datastore_name=f"{customer_name}_input_structured"
-        )
-
-        files_to_blur_txt = Input(
-            type=AssetTypes.URI_FILE,
-            path=files_to_blur_path,
-            description="Data to be blurred",
-        )
-
-        model_parameters = customer['model_parameters']
-        model_parameters_json = json.dumps(model_parameters)  # TODO it seems I can not pass a string to @command_component function
-
-        detect_and_blur_sensitive_data_step = detect_and_blur_sensitive_data(
-            mounted_root_folder=input_root_folder,
-            relative_paths_files_to_blur=files_to_blur_txt,
-            customer_name=customer_name,
-            model_parameters_json=model_parameters_json
-        )
-
-        azureml_outputs_formatted = customer['outputs']['results_path'].format(
-            subscription=subscription_id,
-            resourcegroup=resource_group,
-            workspace=workspace_name,
-            datastore_name=f"{customer_name}_output"
-        )
-
-        detect_and_blur_sensitive_data_step.outputs.results_path = Output(
-            type="uri_folder", mode="rw_mount", path=azureml_outputs_formatted
-        )
-        detect_and_blur_sensitive_data_step.outputs.yolo_yaml_path = Output(
-            type="uri_folder", mode="rw_mount", path=blob_container_path
-        )
+    # Call .result() to get the actual values
+    workspace_name_actual = workspace_name.result()
```

Comment:
Where is this coming from? I don't like that much `result` as a function name, can we name it better?

Reply:
I got weird values when directly calling workspace_name. It has to do with the @pipeline function, this is not a normal python function. I read on internet that its something with lazy loading... Got it from Diana ehe: https://github.com/Computer-Vision-Team-Amsterdam/Blurring-as-a-Service/blob/main/blurring_as_a_service/training_pipeline/submit_training_pipeline.py#L19